src/map/ai/controllers/trust_controller.cpp

```diff
@@ -118,4 +160,48 @@ void CTrustController::DoRoamTick(time_point tick)
             POwner->PAI->PathFind->WarpTo(POwner->PMaster->loc.p, RoamDistance);
         }
     }
+
+	POwner->PAI->EventHandler.triggerListener("TRUST_ROAM_TICK", POwner, POwner->PMaster, PTarget);	
+}
+
+bool CTrustController::Cast(uint16 targid, SpellID spellid)
```

Comment:
if you're overriding these to do exactly what the parent class is doing, then just don't override them and it will call the parent classes one (in other words, remove this and MobSkill)